lib/grape/dsl/parameters.rb

```diff
@@ -199,6 +196,42 @@ def params(params)
         end
         params
       end
+
+      # @param params [Hash] initial hash of parameters
+      # @return hash of parameters with index information relevant for the current scope
+      # example:
+      # params: [{"name"=>"Job", "parents"=>[{"name"=>"Joy"}, {"name"=>"Bob"}], "_param_index": {"children_index": 0}},
+      #   {"name"=>"Jim", "parents"=>[{}], "_param_index": {"children_index": 1}}]
+      # element: parents
+      # result: [{"name"=>"Joy", "_param_index": {children[parents]: 0, "children": 0}},
+      #   {"name"=>"Bob", "_param_index": {children[parents]: 1, "children": 0}},
+      #   { "_param_index": {children[parents]: 0, "children": 1}}]
+      def params_with_index(params)
+        params = @parent.params_with_index(params) if @parent
+        if @element
```

Comment:
Minor: `return params unless @element` or move the `if` to the `end if @element`, less nesting.